event.c

```diff
@@ -383,7 +388,11 @@ static void event_image_snapshot(struct context *cnt, int type ATTRIBUTE_UNUSED,
             return;
         }
     } else {
-        snprintf(fullfilename, PATH_MAX, "%s/lastsnap.%s", cnt->conf.filepath, imageext(cnt));
+		char filename[PATH_MAX];
```

Comment:
So you are declaring those once again just because they are outside of scope? I thing doubling variables of the same name is not a good approach